docs/develop/excel-async-model.md

```diff
@@ -0,0 +1,180 @@
+
+The `context` object is necessary because the script and Excel are running in different processes and locations. The script will need to make changes to or query data from the workbook in the cloud. The `context` object manages those transactions.
+
+By default, Office Scripts handle the interactions between your script and the workbook automatically. While the process is optimized, the standard Office Scripts APIs may synchronize the workbook with your script more than necessary, such as during looped read operations. You may be able to manage these workbook-script data transactions more efficiently with the async APIs, since you know when it is necessary to update data.
+
+## Sync and Load
+
+Because your script and workbook run in different locations, any data transfer between the two takes time. In the async API, commands are queued up until the script explicitly calls the `sync` operation to synchronize the script and workbook. Your script can work independently until it needs to do either of the following:
+
+- Read data from the workbook (following a `load` operation or method that returns a [ClientResult](/javascript/api/office-scripts/excel/excel.clientresult)).
+- Write data to the workbook (usually because the script has finished).
+
+The following image shows an example control flow between the script and workbook:
+
+![A diagram showing read and write operations going to the workbook from the script.](../images/load-sync.png)
+
+### Sync
+
+Whenever your async script needs to read data from or write data to the workbook, call the `RequestContext.sync` method as shown here:
+
+```TypeScript
+await context.sync();
+```
+
+> [!NOTE]
+> `context.sync()` is implicitly called when a script ends.
+
+After the `sync` operation completes, the workbook updates to reflect any write operations that script has specified. A write operation is setting any property on a Excel object (e.g. `range.format.fill.color = "red"`) or calling a method that changes a property (e.g., `range.format.autoFitColumns()`). The `sync` operation also reads any values from the workbook that the script requested by using a `load` operation or a method that returns a `ClientResult` (as discussed in the next sections).
+
+Synchronizing your script with the workbook can take time, depending on your network. You should minimize the number of `sync` calls to help your script run fast. Otherwise, you may as well use the standard, synchronous APIs.
+
+### Load
+
+An async script must load data from the workbook before reading it. However, loading data from the entire workbook would greatly reduce the script's speed. The `load` method lets your script specifically state what data should be retrieved from the workbook.
+
+The `load` method is available on every Excel object. Your script must load an object's properties before it can read them. Not doing so will result in an error.
```

Comment:
```suggestion
The `load` method is available on every Excel object. Your script must load an object's properties before it can read them. Not doing so results in an error.
```